src/interfaces/IStakeManager.sol

```diff
@@ -0,0 +1,26 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.26;
+
+import { IERC20 } from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+
+interface IStakeManager {
+    error StakeManager__FundsLocked();
+    error StakeManager__InvalidLockTime();
+    error StakeManager__InsufficientFunds();
+    error StakeManager__StakeIsTooLow();
+
+    function stake(uint256 _amount, uint256 _seconds) external;
+    function unstake(uint256 _amount) external;
+
+    function totalStaked() external view returns (uint256);
+    function totalMP() external view returns (uint256);
+    function totalMaxMP() external view returns (uint256);
+    function getStakedBalance(address _vault) external view returns (uint256 _balance);
+
+    function STAKE_TOKEN() external view returns (IERC20);
+    function REWARD_TOKEN() external view returns (IERC20);
+    function MIN_LOCKUP_PERIOD() external view returns (uint256);
+    function MAX_LOCKUP_PERIOD() external view returns (uint256);
+    function MP_RATE_PER_YEAR() external view returns (uint256);
+    function MAX_MULTIPLIER() external view returns (uint256);
+}
```

Comment:
@3esmit I left out functions that aren't yet compatible with the reward streamer:

- Upgradeability
- Exiting
- Locking after stake

Will address those in future PRs